pkg/utils/ipnet/ipnet.go

```diff
@@ -114,3 +116,29 @@ func MustParseCIDR(s string) *IPNet {
 	}
 	return cidr
 }
+
+// SplitIntoN splits the parent IPNet into n subnets
+func SplitIntoN(parent *net.IPNet, n int) ([]*net.IPNet, error) {
```

Comment:
Do you want to migrate to this and copy the unit tests from kops? (I think they had a few there).

Reply:
But I'm not sure if needs to be `SplitIntoN`, `SplitInto8` and `SplitInto4` is all we really need. As it must be N^2 , and `SplitInto2` would be of very little use, and I'm not sure of when `SplitInto16` would be applicable...